src/lbaf/Model/lbsBlock.py

```diff
@@ -0,0 +1,76 @@
+import sys
+
+from ..Utils.exception_handler import exc_handler
+
+
+class Block:
+    """ A class representing a memory block with footprint and home
+    """
+    def __init__(
+        self,
+        i: int,
+        h_id: int,
+        size: float=0.0,
+        o_ids: set=set()):
+
+        # Block index
+        if not isinstance(i, int) or isinstance(i, bool):
```

Comment:
Are there missing parentheses here?

Reply:
No because the case for failure is if `i` is not an integer or is a Boolean, because in Python a Boolean is also an integer:
```
>>> isinstance(True, int)
True
>>> 
```
So we want to eliminate such parameter values